src/converse-register.js

```diff
@@ -72,6 +74,16 @@
                             'model': this
                         });
                         this.registerpanel.render().$el.addClass('hidden');
+                        if(_converse.enable_default_domain === true) {
+                            console.log("Registering directly to domain");
+                            var $form = $('form#converse-register');
+                            $form.find('input[name=domain]').attr('disabled', 'disabled');
+                            console.log(_converse.default_domain_name);
+                            $form.find('input[name=domain]').val(_converse.default_domain_name);
+                            console.log($form.find('input[name=domain]').val());
+                            $form.trigger("submit_auto");
+                            console.log("Rendered registration form");
```

Comment:
I don't like this way of doing things.

Instead of rendering a form and then programmatically setting values and submitting it, you can just avoid all that and fetch the registration form yourself in the code.

Something like this:

```
if(!._.isEmpty(_converse.registration_domain)) {
    chooseProvider();
}
```

Then `chooseProvider` looks basically like the last 5 lines of `onProviderChosen`. In fact, you can then also let `onProviderChosen` call `chooseProvider` and remove those 5 lines from it.